File: array/DutchNationalFlagAlgo.java
package array;
public class DutchNationalFlagAlgo {
    public static void dutchNationalFlagAlgorithm(int[] nums){
        int n = nums.length;
        int low = 0;
        int mid = 0;
        int high = n-1;
        while(mid <= high){
            if(nums[mid] == 0){
                int temp = nums[mid];
                nums[mid] = nums[low];
                nums[low] = temp;
                low++;
                mid++;
            }else if(nums[mid] == 1) {
                mid++;
            }else{
                int temp = nums[mid];
                nums[mid] = nums[high];
                nums[high] = temp;
                high--;
            }
        }
    }

    public static void printArray(int[] arr){
        for(int i =0;i<arr.length;i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = {2,0,1};
        dutchNationalFlagAlgorithm(arr);
        printArray(arr);
    }
}

File: array/dutch_national_flag_algo.js
function dutch_nationa_flag_algo(nums){
    let n = nums.length;
    let low =0;
    let mid =0;
    let high = n-1;
    while(mid<=high){
        if(nums[mid] ==1){
            mid++;
        }else if(nums[mid] == 0){
            let temp = nums[mid];
            nums[mid] = nums[low];
            nums[low] = temp;
            low++;
            mid++;
        }else{
            let temp = nums[mid];
            nums[mid] = nums[high];
            nums[high] = temp;
            high--;
        }
    }
}


let arr = [1,1,2,2,0];
dutch_nationa_flag_algo(arr);
console.log(arr)